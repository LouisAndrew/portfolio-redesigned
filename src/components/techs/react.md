---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
