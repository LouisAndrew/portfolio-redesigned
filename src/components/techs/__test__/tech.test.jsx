import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Tech from '../tech'

describe('tech small component.', () => {

        afterEach( cleanup )

        const techName = 'ReactJS'
        const techNameBroken = 'broken'
        const click = () => 'clicked!'

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(<Tech techName={techName} />, div)
        })

        it('renders even when the provided techName is not valid', () => {

                const div = document.createElement('div')
                ReactDOM.render(<Tech techName={techNameBroken} />, div)
        })

        it('renders correctly', () => {

                const { getByTestId } = render( <Tech techName={techName} onClick={click} /> )
                const techComponent = getByTestId('tech')
        })

        it('matches snapshot', () => {

                const tree = renderer.create( <Tech techName={techName} onClick={click} /> ).toJSON()
                expect(tree).toMatchSnapshot()
        })

})