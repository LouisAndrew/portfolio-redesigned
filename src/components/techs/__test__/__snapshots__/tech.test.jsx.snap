// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tech small component. matches snapshot 1`] = `
<div
  className="sc-AxjAm krZeBP"
  data-testid="tech"
  onClick={[Function]}
/>
`;
